### ISSUE TITLE
Query - reconciling existing tests with problem-specifications

### ISSUE THREAD

**Issue by bencoman** [exercism/pharo-smalltalk#377]
@macta, To validate my refactoring of generation, I'm comparing existing tests to what "would" be generated.  There are some anomalies that may just need some background info.
To try to keep things organised, I'm going to put a query per post.  Could you <EDIT> each of my posts if you have any comment.

**Comment by bencoman** [comment]
GradeSchoolTest>>#test100_EnsureDataIsImmutable
doesn't have a corresponding problem-specification 
and its jump to number 100 indicates it was a manual entry?

**Comment by bencoman** [comment]
RobotSimulatorTest>>test15 (to test18)
    WhereRTurnRightLTurnLeftAndAAdvance....
    WhereREqualsTurnRightLEqualsTurnLeftAndAEqualsAdvance....
is different when dealing with "equals" sign

Bypassing this difference by cleaning "descriptions" in problem-specifications 
https://github.com/exercism/problem-specifications/pull/1525

**Comment by bencoman** [comment]
LeapTest>>#test01_YearNotDivisibleBy4CommonYear
and all the rest of the methods are missing the "In" from "InCommonYear".
I'd guess this is a legit difference due to an updated description... https://github.com/exercism/problem-specifications/commit/18875ece112eb46c7227e7e1745432ed16030ddd

**Comment by bencoman** [comment]
EtlTest>>#test01_TransformsASingleLetter
in my name mapping becomes... 
test....TransformsTheASetOfScrabbleDataPreviouslyIndexedByTheTileScoreToASetOfDataIndexedByTheTileLetterASingleLetter

Was this super big method name by chance manually slimmed down?
Anyway, I'm fixing the  problem-specification to match the current method name... https://github.com/exercism/problem-specifications/pull/1526

**Comment by bencoman** [comment]
ClockTest>>#test27_AddMinutesAddMoreThanOneDay1500Min25Hrs
compares to...         _AddMinutesAddMoreThanOneDay1500MinEquals25Hrs

So again a difference due to the conversion of ""=" to "equals".
Was this a late addition? Or at lesat after ClockTest was generated?

**Comment by bencoman** [comment]
`Allergies` is strange.  In image are only 20 test methods but the problem specification has 49 tests.
I can match about half of the test methods to problem-specifications,  but the rest leaves me stumped.

**Comment by bencoman** [comment]
AnagramTest>>#test10_DoesNotDetectAnAnagramIfTheOriginalWordIsRepeated
has an "An" versus "DoesNotDetectAAnagramIfTheOriginalWordIsRepeated 
generated from problem-specification... https://github.com/exercism/problem-specifications/blob/master/exercises/anagram/canonical-data.json#L94
So was that manually corrected?

**Comment by macta** [comment]
Yes - its in the category “tests-extra” ( I wondered if regenerating might be able preserve any extra tests we might feel are suitable?)

Tim

**Comment by macta** [comment]
I think you’ve fully understood the problem now ;). - the definitions are a bit all over the place. I’ve managed to get a few upstream corrections approved but there are quite a few of them.

**Comment by macta** [comment]
Yes, I proposed an upstream change

**Comment by macta** [comment]
I don’t recall this one - It does like it was manually adjusted, but I don’t recall doing that (and have tried not to and either made a pr upstream or improved the generator).

**Comment by macta** [comment]
Clock was an awkward one - hence https://github.com/exercism/pharo-smalltalk/issues/246 <https://github.com/exercism/pharo-smalltalk/issues/246> and some discussions on exercism problem spec.

I did transformations using the deprecation tool, but it was laborious and unsatisfying. I think there should be a meta hint that this is an equality test and then the generator could do the right thing (and we would have some equality generation policy).

**Comment by macta** [comment]
Allergies, I made a pr upstream as it was a mess - this caused lots of discussion and change - the generated version is 1.2.1 its now at 2, so this is where regeneration would be helpful (we detect a higher version and then generate on top - hopefully in a diffable way?)

**Comment by macta** [comment]
I think this one was user submitted by Ray, so he might have corrected it without doing a pr upstream.

**Comment by macta** [comment]
I think you’ve understood all of these (with the added info about the #generator method)
